### ISSUE TITLE
Caught exceptions delete captured parameters

### ISSUE THREAD

**Issue by ahawkins** [sinatra/sinatra#860]
Found this during my work today. Not sure if this is a bug or feature. If this is a feature, I'm keen to know why. Assume you have a route like `GET /users/:id`. Inside the router handler an error occurs and is caught by an `error` block. `params['id']` is not there and `params['captures']` returns the error instance. Here is an executable test case to describe the problem:

```ruby
ENV['RACK_ENV'] = 'test'
require 'sinatra'
require 'minitest/autorun'
require 'rack/test'

class ErrorCaptureTest < MiniTest::Unit::TestCase
  include Rack::Test::Methods

  TestError = Class.new StandardError

  attr_reader :app

  def test_rewrites_capatured_params
    @app = Class.new Sinatra::Base do
      error TestError do
        halt 200, params.fetch('name', 'blank')
      end

      get '/:name' do
        fail TestError
      end
    end

    get '/adam'

    assert_equal 'adam', last_response.body
  end
end
```

**tl;dr Expected behavior:** captured params be accessible inside error blocks.
